src/Metadata/FilterInterface.php

```diff
@@ -13,59 +13,47 @@
 
 namespace ApiPlatform\Metadata;
 
-if (interface_exists(\ApiPlatform\Api\FilterInterface::class)) {
-    class_alias(
-        \ApiPlatform\Api\FilterInterface::class,
-        __NAMESPACE__.'\FilterInterface'
-    );
-
-    if (false) { // @phpstan-ignore-line
-        interface FilterInterface extends \ApiPlatform\Api\FilterInterface
-        {
-        }
-    }
-} else {
+/**
+ * Filters applicable on a resource.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+interface FilterInterface
+{
     /**
-     * Filters applicable on a resource.
+     * Gets the description of this filter for the given resource.
+     *
+     * Returns an array with the filter parameter names as keys and array with the following data as values:
+     *   - property: the property where the filter is applied
+     *   - type: the type of the filter
+     *   - required: if this filter is required
+     *   - description (optional) : the description of the filter
+     *   - strategy (optional): the used strategy
+     *   - is_collection (optional): if this filter is for collection
+     *   - swagger (optional): additional parameters for the path operation,
+     *     e.g. 'swagger' => [
+     *       'description' => 'My Description',
+     *       'name' => 'My Name',
+     *       'type' => 'integer',
+     *     ]
+     *   - openapi (optional): additional parameters for the path operation in the version 3 spec,
+     *     e.g. 'openapi' => [
+     *       'description' => 'My Description',
+     *       'name' => 'My Name',
+     *       'schema' => [
+     *          'type' => 'integer',
+     *       ]
+     *     ]
+     *   - schema (optional): schema definition,
+     *     e.g. 'schema' => [
+     *       'type' => 'string',
+     *       'enum' => ['value_1', 'value_2'],
+     *     ]
+     * The description can contain additional data specific to a filter.
+     *
+     * @see \ApiPlatform\OpenApi\Factory\OpenApiFactory::getFiltersParameters
      *
-     * @author Kévin Dunglas <dunglas@gmail.com>
+     * @return array<string, array{property: ?string, type: string, required: bool, description?: string, strategy?: string, is_collection?: bool, swagger?: array<string, mixed>, openapi?: array<string, mixed>|\ApiPlatform\OpenApi\Model\Parameter, schema?: array<string, mixed>}>
```

Comment:
Mhh my bad I think that `?string` is wrong, I thought that everything was optional but let's keep that. We may need to change the tests you mention, there shouldn't be lots of them right? 

```suggestion
     * @return array<string, array{property: string, type: string, required: bool, description?: string, strategy?: string, is_collection?: bool, swagger?: array<string, mixed>, openapi?: array<string, mixed>|\ApiPlatform\OpenApi\Model\Parameter, schema?: array<string, mixed>}>
```

Reply:
Rollback done.